src/classes/GiftBatch.cls

```diff
@@ -85,6 +99,20 @@ public inherited sharing class GiftBatch {
 
     public GiftBatch(Id batchId) {
         this.batchId = batchId;
+        this.batch = giftBatchSelector.selectGiftBatchById(this.batchId);
+    }
+
+    public Boolean hasElevateWidget() {
+        Boolean hasWidget = false;
+        for (GE_Template.Section section : formTemplate.layout.sections) {
+            for (GE_Template.Element element : section.elements) {
+                if (element.componentName != null
+                    && element.componentName == 'geFormWidgetTokenizeCard') {
```

Comment:
_Nice to have (non-blocking)_

We could extract the guard clause into a `isElevateWidget(element)` method.

Reply:
Updated!